cln-rpc/src/primitives.rs

```diff
@@ -254,6 +254,41 @@ pub enum ChannelSide {
     REMOTE,
 }
 
+impl TryFrom<i32> for ChannelSide {
+    type Error = crate::Error;
+
+    fn try_from(value: i32) -> std::result::Result<Self, Self::Error> {
+        match value {
+            0 => Ok(ChannelSide::LOCAL),
+            1 => Ok(ChannelSide::REMOTE),
+            _ => Err(anyhow!(
+                "Invalid ChannelSide mapping, only 0 or 1 are allowed"
+            )),
+        }
+    }
+}
+
+impl TryFrom<i32> for ChannelState {
+    type Error = crate::Error;
+
+    fn try_from(value: i32) -> std::result::Result<Self, Self::Error> {
+        match value {
+            0 => Ok(ChannelState::OPENINGD),
+            1 => Ok(ChannelState::CHANNELD_AWAITING_LOCKIN),
+            2 => Ok(ChannelState::CHANNELD_NORMAL),
+            3 => Ok(ChannelState::CHANNELD_SHUTTING_DOWN),
+            4 => Ok(ChannelState::CLOSINGD_SIGEXCHANGE),
+            5 => Ok(ChannelState::CLOSINGD_COMPLETE),
+            6 => Ok(ChannelState::AWAITING_UNILATERAL),
+            7 => Ok(ChannelState::FUNDING_SPEND_SEEN),
+            8 => Ok(ChannelState::ONCHAIN),
+            9 => Ok(ChannelState::DUALOPEND_OPEN_INIT),
+            10 => Ok(ChannelState::DUALOPEND_AWAITING_LOCKIN),
+            _ => Err(anyhow!("Invalid channel state {}", value)),
```

Comment:
This will likely require us to explicitly annotate the `enum` declaration, otherwise changing the order in the declaration would result in a mismatch here.

Reply:
I added https://github.com/ElementsProject/lightning/pull/5890/commits/7a829a5db4ae6ef6f45c42acaec5509a0301e17b

IIUC I think the issue wasn't introduced by this code, since the mapping is defined in the proto files https://github.com/ElementsProject/lightning/blob/9482e0619c005bb183e4d120844b9ade545fdd0a/cln-grpc/proto/primitives.proto#L27 and there is this conversion code https://github.com/ElementsProject/lightning/blob/9482e0619c005bb183e4d120844b9ade545fdd0a/cln-grpc/src/convert.rs#L159 
any reordering would have caused an issue in master

Reply:
Absolutely agree, the original mistake was on my side, and this was more of a drive-by thing we can address. Sorry for not being clear about that here.